src/frontend/context/SettingsContext.tsx

```diff
@@ -54,12 +53,30 @@ export const SettingsProvider = ({ children }: React.PropsWithChildren<{}>) => {
     [setState]
   );
 
+  /**
+   * @typedef {SettingsContextType[2]} setExperiment
+   * @param key The key of the experiment to toggle
+   * @param value The value to set the experiment to. If not provided the value will be toggled
+   * @description Sets the value of the experiment to the provided value or toggles it if no value is provided
+   */
+  const setExperiments: SettingsContextType[2] = React.useCallback(
```

Comment:
just so i'm understanding this, this is so that you can toggle experiments more granularly, as opposed to having to work at the top-level settings?

Reply:
suggestion: move this to `useExperiments` hook, unless there are places that aren't using the hook and instead using this context directly

Reply:
nit: the `set` in the naming feels a little unclear when actually using this for toggling a value. calling this via `setExperiments("name_of_experiment")` doesn't give a clear idea of _what_ i'm setting it to. personally try to avoid functions that just toggle state because of this lack of clarity when being called

Reply:
I addressed the explicit settings boolean, by making it a mandatory parameter

Reply:
Moved the logic to the hook